File: Data Structures & Algorithms/max-consecutive-ones/submission-0.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        //create & initalize counter variables
        int current = 0;
        int max = 0;

        for(int i = 0; i < nums.length; i++){

            if(nums[i]==1){
                current++; //increment current counter
            }
            else{
                current = 0; //reset current counter
            }
            if(current > max){
                max = current; //update max to current if > existing max
            }
        }
        return max;
    }
}

File: Data Structures & Algorithms/maximum-ascending-subarray-sum/submission-0.py
class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        res = nums[0]
        curr_sum = nums[0]
        for i in range(1,len(nums)):
            if nums[i] <= nums[i-1]:
                curr_sum = 0
            
            curr_sum += nums[i]
            res = max(res, curr_sum)

        return res

            
        